packages/edit-navigation/src/store/test/resolvers.js

```diff
@@ -28,288 +37,163 @@ jest.mock( '@wordpress/blocks', () => {
-						},
-					],
-					name: 'core/navigation',
-				},
-			],
-			meta: {
-				menuId,
-			},
-		};
+	it( 'returns early when a menuId is not provided', async () => {
+		const menuId = null;
 
-		expect( generator.next( menuItems ).value ).toEqual(
-			dispatch(
-				'core',
-				'receiveEntityRecords',
-				NAVIGATION_POST_KIND,
-				NAVIGATION_POST_POST_TYPE,
-				navigationBlockStubPost,
-				{ id: navigationBlockStubPost.id },
-				false
-			)
-		);
+		const registry = createRegistryWithStores();
+		await registry
+			.resolveSelect( editNavigationStore )
+			.getNavigationPostForMenu( menuId );
 
-		expect( generator.next().value ).toEqual(
-			dispatch( 'core', 'finishResolution', 'getEntityRecord', [
-				NAVIGATION_POST_KIND,
-				NAVIGATION_POST_POST_TYPE,
-				stubPost.id,
-			] )
-		);
+		const stubPost = registry
+			.select( coreStore )
+			.getEntityRecord( 'root', 'postType', menuId );
 
-		expect( generator.next().done ).toBe( true );
+		expect( stubPost ).toBeFalsy();
 	} );
 
-	it( 'creates correct core/navigation-link block variations from menu objects', () => {
+	it( 'creates a stub navigation post for menu id', async () => {
 		const menuId = 123;
 
-		const generator = getNavigationPostForMenu( menuId );
+		const registry = createRegistryWithStores();
+		const stubPost = await registry
+			.resolveSelect( editNavigationStore )
+			.getNavigationPostForMenu( menuId );
 
-		// Advance generator
-		generator.next(); // Gen step: yield persistPost
-		generator.next(); // Gen step: yield dispatch "getEntityRecord"
-		generator.next(); // Gen step: yield resolveMenuItems
+		expect( stubPost ).toBeTruthy();
+		expect( stubPost.id ).toBe( 'navigation-post-123' );
+		expect( stubPost.blocks ).toHaveLength( 1 );
+		expect( stubPost.blocks[ 0 ].innerBlocks ).toHaveLength( 3 );
+	} );
 
-		const menuItems = [
-			{
-				id: 100,
-				title: {
-					raw: 'wp.com',
-					rendered: 'wp.com',
-				},
-				url: 'http://wp.com',
-				menu_order: 1,
-				menus: 1,
-				parent: 0,
-				classes: [ 'menu', 'classes' ],
-				xfn: [ 'nofollow' ],
-				description: 'description',
-				attr_title: 'link title',
-				object: 'post',
-				object_id: 123, // the post object ID not the menu object ID
-			},
-			{
-				id: 101,
-				title: {
-					raw: 'wp.org',
-					rendered: 'wp.org',
-				},
-				url: 'http://wp.org',
-				menu_order: 2,
-				menus: 1,
-				parent: 0,
-				classes: [],
-				xfn: [],
-				description: '',
-				attr_title: '',
-				object: 'page',
-				object_id: 456, // the page object ID not the menu object ID
-			},
-			{
-				id: 102,
-				title: {
-					raw: 'wordpress.org',
-					rendered: 'wordpress.org',
-				},
-				url: 'https://wordpress.org',
-				menu_order: 3,
-				menus: 1,
-				parent: 0,
-				classes: [],
-				xfn: [],
-				description: '',
-				attr_title: '',
-				object: 'custom',
-			},
-		];
+	it( 'creates correct core/navigation-link block variations from menu objects', async () => {
```

Comment:
This test is the same as the one before, no? Just with different assertions? Can we combine them?